src/components/EarningAreaChart/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { useState } from 'react'
+import { isMobile } from 'react-device-detect'
+import { Area, AreaChart, Customized, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts'
+
+import { EMPTY_FUNCTION } from 'constants/index'
+import useTheme from 'hooks/useTheme'
+import { TimePeriod } from 'pages/MyEarnings/MyEarningsOverTimePanel/TimePeriodSelect'
+import KyberLogo from 'pages/TrueSightV2/components/chart/KyberLogo'
+import { EarningStatsTick } from 'types/myEarnings'
+import { toFixed } from 'utils/numbers'
+
+import TooltipContent from './TooltipContent'
+import { formatUSDValue } from './utils'
+
+const labelGapByTimePeriod: Record<TimePeriod, number> = {
+  ['7D']: isMobile ? 2 : 1,
+  ['1M']: isMobile ? 8 : 3,
+  ['6M']: isMobile ? 40 : 20,
+  ['1Y']: isMobile ? 90 : 30,
+}
+
+const CustomizedLabel = (props: any) => {
+  const theme = useTheme()
+  const { x, y, value, index, period } = props
+  const show = (index + 1) % (labelGapByTimePeriod[period as TimePeriod] || 1) === 0
+  return (
+    <>
+      {show && (
+        <text
+          x={x}
+          y={y}
+          dx={index ? undefined : 20}
+          dy={-10}
+          fontSize={12}
+          fontWeight={500}
+          fill={theme.subText}
+          textAnchor="middle"
+        >
+          {formatUSDValue(value)}
+        </text>
+      )}
+    </>
+  )
+}
+
+const subscriptMap: { [key: string]: string } = {
```

Comment:
```
arr = [ '₀', '₁', '₂',  ...]
```

Reply:
I prefer map in this case

Reply:
If You dont have any debate, I will resolve conversation =))